src/layer/vector/Canvas.js

```diff
@@ -145,6 +152,8 @@ export var Canvas = Renderer.extend({
 			layer._path.setAttribute('aria-label', layer.options.title);
 			layer._path.setAttribute('title', layer.options.description);
 
+
+			DomEvent.on(layer._path, 'keypress', Util.bind(this._onKeyPress, this, layer));
```

Comment:
I notice that this does not work when using a screen reader (tested in Narrator and NVDA). Similar to https://github.com/Leaflet/Leaflet/pull/8202/commits/381268965b3badb0f34ae6eb446660026af2dadd.

Reply:
Screen readers have a tendency to intercept keyboard events, not exactly sure under which conditions though.

Other cases where things work(ed) well without a screen reader, but doesn't if you are using one (may be totally unrelated):
- Popups only opening on <kbd>enter</kbd> if markers have `role="button"`: https://github.com/Leaflet/Leaflet/pull/7895#issue-1097336107
- https://github.com/Leaflet/Leaflet/issues/8114

Reply:
Yes, I need to rethink this part and make it more robust. Unfortunately, I can not test things with Narrator or NVDA. But I currently can think of 2 possible reasons for this:
- Screen readers send may some other events (like a fake `click`), but I pass through only `keypress`. I will change that and pass other events too.
- Maybe it's related to the role being `graphics element` and not telling the user they can interact with it, as mentioned in the linked issue (#7895). If that's the case, #8250 should take care of that.